pkg/lib/limits.go

```diff
@@ -0,0 +1,67 @@
+package telemetrylib
+
+import (
+	"errors"
+	"log"
+)
+
+const (
+	// 5MB
+	TELEMETRY_DATA_MAX_SIZE = 5242880
+	TELEMETRY_DATA_MIN_SIZE = 10
+)
+
+type TelemetryDataLimits struct {
+	MaxSize uint64
+	MinSize uint64
+}
+
+// func NewTelemetryDataLimits(data []byte) *TelemetryDataLimits {
+func NewTelemetryDataLimits(data []byte) (*TelemetryDataLimits, error) {
```

Comment:
See comments above about separating this into two helper functions, namely:
* CheckTelemetryDataLimits([]bytes) error
* NewTelemetryDataLimits() *TelemetryDataLimits